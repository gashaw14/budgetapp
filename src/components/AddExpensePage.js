import React from 'react';

const AddExpensePage = ()=>(
    <p>add expenses here</p>
)
export default AddExpensePage;