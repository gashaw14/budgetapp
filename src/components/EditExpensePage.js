import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import Modal from 'react-modal';


export class EditExpensePage extends React.Component {
    state = {
        modalOpen: false
    }
    delete = () => {
        this.setState(() => ({
            modalOpen: false
        }))
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push('/dashboard');
    }

    noDelete = () => {
        this.setState(() => ({
            modalOpen: false
        }))
        this.props.history.push('/dashboard');
    }
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/dashboard');
    }

    onRemove = () => {
        this.setState(() => ({
            modalOpen: true
        }))

        // this.props.startRemoveExpense({ id: this.props.expense.id });
        // this.props.history.push('/dashboard');
    }
    render() {
        return (
            <div >
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />
                    <button className="button button--secondary" onClick={this.onRemove}>Remove Expense</button>

                </div>
                
    
                    <Modal
                        isOpen={this.state.modalOpen}
                        onRequestClose={this.delete || this.noDelete}
                        closeTimeoutMS={200}
                        className="modal"

                    >
                        <p className="modal__title">Are you sure to delete your expense?</p>
                        <button className=" button button--modal" onClick={this.noDelete}>No</button>
                        <button className="button button--modal" onClick={this.delete}>Yes</button>
                    </Modal>
                   
                   
                </div>

        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
                    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
        startRemoveExpense: (data) => dispatch(startRemoveExpense(data))

    }

}

const mapStateToProps = (state, props) => {
    return {
                    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    }
}


export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);