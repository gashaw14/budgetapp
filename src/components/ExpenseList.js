import React from 'react';
import { connect } from 'react-redux';
import {getVisibileExpenses} from '../selectors/expenses';
import ExpenseListItem from './ExpenseListItem';

const ExpenseList = (props)=>{
    return(
        <div>
            <h1>Expense List</h1>
         {props.expenses.map((expense)=>{
          return <ExpenseListItem key={expense.id}{...expense}/>
         })}   
        </div>
    )
}

const mapStateToProps = (state)=>{
return {
expenses: getVisibileExpenses(state.expenses,state.filters)
}
}
export default connect(mapStateToProps)(ExpenseList);
