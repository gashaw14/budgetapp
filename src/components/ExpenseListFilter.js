import React from 'react';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';

export class ExpenseListFiter extends React.Component {

    state = {
        focusedInput: null
    }


    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    }
    onFocusChange = (focusedInput) => {
        this.setState(() => ({ focusedInput }))
    }

    onTextValueChange = (e) => {
        if (e.target.value === 'amount') {
            this.props.sortByAmount();
        } else if (e.target.value === 'date') {
            this.props.sortByDate()
        }
    }
    onTextInputChange = (e) => {
        this.props.setTextFilter(e.target.value)
    }

    render() {
        return (
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input type='text'
                            className="text-input"
                            placeholder="Search expenses"
                            value={this.props.filters.text}
                            onChange={this.onTextInputChange} />
                    </div>
                    <div className="input-group__item">
                        <select className="select"
                            value={this.props.filters.sortBy}
                            onChange={this.onTextValueChange}
                        >
                            <option value='amount'>Amount</option>
                            <option value='date'>Date</option>
                        </select>
                    </div>
                    <div className="input-group__item">
                        <DateRangePicker
                            startDate={this.props.filters.startDate}
                            endDate={this.props.filters.endDate}
                            onDatesChange={this.onDatesChange}
                            focusedInput={this.state.focusedInput}
                            onFocusChange={this.onFocusChange}
                            numberOfMonths={1}
                            showClearDates={true}
                            isOutsideRange={() => false}
                        />
                    </div>
                </div>
            </div>
        )
    }
}
const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }

}

const mapDispatchToProps = (dispatch) => ({
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate)),
    sortByAmount: () => dispatch(sortByAmount()),
    sortByDate: () => dispatch(sortByDate()),
    setTextFilter: (data) => dispatch(setTextFilter(data))

})
export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFiter)