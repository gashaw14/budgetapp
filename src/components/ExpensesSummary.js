import React from 'react';
import {connect} from 'react-redux';
import numeral from 'numeral';
import {getVisibileExpenses} from '../selectors/expenses';
import expensesTotal from '../selectors/Expenses-total';

export const ExpensesSummary = ({expenseCount, amountCount})=>{
    const expenseWord = expenseCount ===1 ? 'expense':'expenses'
    const dollar = numeral(amountCount/100).format('$0,0.00');
    return(
        <div>
        <h1> you have {expenseCount} {expenseWord} with {dollar} total amount</h1>
        </div>
    )
}

const mapSateToProps = (state)=>{
   const visibleExpenses = getVisibileExpenses(state.expenses, state.filters);

   return{
       expenseCount: visibleExpenses.length,
       amountCount: expensesTotal(visibleExpenses)
   }
}

export default connect(mapSateToProps)(ExpensesSummary);