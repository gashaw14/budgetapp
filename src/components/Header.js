import React from 'react';
import {NavLink } from 'react-router-dom';
import {connect} from 'react-redux';
import {startLogout} from '../actions/auth';


export const Header = ({startLogout}) =>{
   return(
       <div>
       <h1>Budget App</h1>
           <NavLink to="/dashboard" activeClassName="is-active" >DashBoard</NavLink>
           <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink>
           <button onClick={startLogout}>Logout</button>
       </div>
   )       
}

const mapDispatchToProps = (dispatch)=>{
    return{
        startLogout: ()=>dispatch(startLogout())
    }
}
export default connect(undefined, mapDispatchToProps)(Header);