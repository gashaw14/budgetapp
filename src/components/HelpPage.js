import React from 'react';

const HelpPage =()=>(
<p>help page</p>
)
    


export default HelpPage;