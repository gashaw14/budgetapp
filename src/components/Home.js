import React from 'react';

const Home =()=>(
    <h1>home page</h1>
)
   


export default Home;