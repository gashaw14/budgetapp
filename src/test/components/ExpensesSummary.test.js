import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';

test('should test a single expense summary',()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={1} amountCount={200}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should test a multiple expenses summary',()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={3} amountCount={23000000}/>);
    expect(wrapper).toMatchSnapshot();
});