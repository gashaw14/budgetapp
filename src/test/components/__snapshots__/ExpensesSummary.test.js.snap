// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test a multiple expenses summary 1`] = `
<div>
  <h1>
     you have 
    3
     
    expenses
     with 
    $230,000.00
     total amount
  </h1>
</div>
`;

exports[`should test a single expense summary 1`] = `
<div>
  <h1>
     you have 
    1
     
    expense
     with 
    $2.00
     total amount
  </h1>
</div>
`;
